plugins/woocommerce/src/Internal/Admin/Settings/PaymentProviders/WooPayments/WooPaymentsService.php

```diff
@@ -1441,15 +1443,27 @@ private function get_onboarding_steps( string $location, string $rest_path ): ar
 		// If the WPCOM connection is already set up, we don't need to add anything more.
 		if ( self::ONBOARDING_STEP_STATUS_COMPLETED !== $wpcom_step['status'] ) {
 			// Craft the return URL.
-			// By default, we return the user to the onboarding modal.
-			$return_url = $this->proxy->call_static(
-				Utils::class,
-				'wc_payments_settings_url',
-				self::ONBOARDING_PATH_BASE,
-				array(
-					'wpcom_connection_return' => '1', // URL query flag so we can properly identify when the user returns.
-				)
-			);
+			switch ( $source ) {
+				case 'launch-your-store':
+					// If the source is 'launch-your-store', we return the user to the Launch Your Store flow.
+					$return_url = $this->proxy->call_function(
```

Comment:
@vladolaru, out of curiosity, what's the difference between defining URLs like this versus what we already had? Is it safe to define it this way?